reporter/src/main/kotlin/model/EvaluatedModel.kt

```diff
@@ -19,7 +19,284 @@
+    var paths: MutableList<EvaluatedPackagePath> = mutableListOf(),
+    val levels: SortedSet<Int>,
+    val scopes: MutableSet<ScopeName>,
+    val scanResults: List<EvaluatedScanResult>,
+    val findings: List<EvaluatedFinding>,
+    val isExcluded: Boolean,
+    val pathExcludes: List<PathExclude>,
+    val scopeExcludes: List<ScopeExclude>,
+    val issues: List<EvaluatedOrtIssue>
+)
+
+data class EvaluatedPackagePath(
+    val pkg: EvaluatedPackage,
+    val project: EvaluatedPackage,
+    val scope: ScopeName,
+    val path: List<EvaluatedPackage>
+)
+
+data class EvaluatedScanResult(
+    val provenance: Provenance,
+    val scanner: ScannerDetails,
+    val startTime: Instant,
+    val endTime: Instant,
+    val fileCount: Int,
+    val packageVerificationCode: String,
+    val issues: List<EvaluatedOrtIssue>
+)
+
+enum class EvaluatedFindingType {
+    COPYRIGHT, LICENSE
+}
+
+data class EvaluatedFinding(
+    val type: EvaluatedFindingType,
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val license: LicenseId?,
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val copyright: CopyrightStatement?,
+    val path: String,
+    val startLine: Int,
+    val endLine: Int,
+    val scanResult: EvaluatedScanResult
+)
+
+data class CopyrightStatement(
+    val statement: String
+)
+
+data class LicenseId(
+    val id: String
+)
+
+data class ScopeName(
+    val name: String
+)
+
+data class EvaluatedProcessedDeclaredLicense(
+    val spdxExpression: SpdxExpression?,
+    val mappedLicenses: List<LicenseId>,
+    val unmappedLicenses: List<LicenseId>
+)
+
+enum class EvaluatedOrtIssueType {
+    ANALYZER, SCANNER
+}
+
+@JsonIgnoreProperties(value = ["pkg", "scanResult", "path"], allowGetters = true)
+data class EvaluatedOrtIssue(
+    val timestamp: Instant,
+    val type: EvaluatedOrtIssueType,
+    val source: String,
+    val message: String,
+    val severity: Severity = Severity.ERROR,
+    val resolutions: List<IssueResolution>,
+    @JsonIdentityReference(alwaysAsId = true)
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    var pkg: EvaluatedPackage? = null,
+    @JsonIdentityReference(alwaysAsId = true)
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    var scanResult: EvaluatedScanResult? = null,
+    @JsonIdentityReference(alwaysAsId = true)
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val path: EvaluatedPackagePath? = null
+)
+
+data class EvaluatedRuleViolation(
+    val rule: String,
+    val pkg: EvaluatedPackage,
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val license: LicenseId? = null,
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val licenseSource: LicenseSource? = null,
+    val severity: Severity,
+    val message: String,
+    val howToFix: String,
+    val resolutions: List<RuleViolationResolution>
+)
+
+@JsonIdentityInfo(property = "key", generator = ZeroBasedIntSequenceGenerator::class, scope = DependencyTreeNode::class)
+data class DependencyTreeNode(
+    val title: String,
```

Comment:
Where does the `title` property come from?
What is it supposed to be used for?

Reply:
It's the title of the node, depending on what it represents it can be the ID of a project, the name of a scope, or the ID of a package.

Reply:
Ok, so my understanding now is that this can either represent a `package`, `project` or `scope` 
and depending on what it actually is `nullness / empty implications` apply to some of the other properties.

Looking at this kind of design decisions throughout this change am I actually right to assume that the model is heavily tailored towards the web-app and maybe not meant to be used by other reporters or even the evaluator in the future?

Reply:
I see this as an iterative process, the first version is tailored towards the web app reporter. Once this is done using it for other reporters will show how well that works. I think one possible outcome will be that some of the optimizations done for the web app reporter will be moved from this model to a separate optimization step.